client/my-sites/activity/filterbar/date-range-selector.jsx

```diff
@@ -162,7 +162,10 @@ export class DateRangeSelector extends Component {
 								className="filterbar__selection-close"
 								compact
 								borderless
-								onClick={ this.handleResetSelection }
+								onClick={ () => {
+									this.handleResetSelection();
+									props.onClearClick();
```

Comment:
I think it would be better to move the `onClick` handler out of the JSX to avoid creating a new function on every render. Maybe we could call `props.onClearClick()` inside the `handleResetSelection` function. What do you think?

Reply:
Funny! My previous approach was to pass `props.onClearClick()` to  `handleResetSelection` 😄  

https://github.com/Automattic/wp-calypso/pull/92454/commits/781524c01f4e24b81a41e2ea387d588a91879dda

After doing it, I felt I was coupling too much `date-range-selector` (filterbar from activity) with the [date-range](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/date-range/index.js) component, and that maybe the function `handleResetSelection` shouldn't need to know about that function (anyway I was creating a new function)

What should be the code to get the onClick handler out of the JSX? 🤔

Reply:
> What should be the code to get the onClick handler out of the JSX? 🤔

The idea is to not declare the function directly in the JSX like this:

```
onClick={ () => {
    this.handleResetSelection();
    props.onClearClick();
}
```

Doing that, a new function will be created on every render. To avoid that, we could declare it as part of the component and then just pass the function into the JSX (similar to what the code looked like before the change):

```

onClickHandler = () => {
    this.handleResetSelection();
    props.onClearClick();
};

...
<Button
	className="filterbar__selection-close"
	compact
	borderless
	onClick={ this.onClickHandler }
>
...
```